### PR TITLE
Add hard swish op (new op)

### PR DESCRIPTION
**Pull request by huangjun12**
add new op hard swish. It is an activation op.

### CODE REVIEW

File: my_API.spec [hunk 1]
```diff
@@ -0,0 +1,1047 @@
+paddle.fluid.Program ('paddle.fluid.framework.Program', ('document', '7364a01d7b9132a435e46162c7fbd6c6'))
```
Comment (shippingwang) [inline my_API.spec]: remove this file.

**Review by shippingwang — CHANGES_REQUESTED** [review]
and remove ".nn.py.swp"

**Comment by huangjun12** [comment]
image

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -12811,3 +12812,38 @@ def shard_index(input, index_num, nshards, shard_id, ignore_value=-1):
...
         stop_gradient=True)
     return out
+
+
+@templatedoc()
+def hard_swish(x, threshold=6.0, scale=6.0, offset=3.0, name=None):
+    """
+    ${comment}
+    Args:
+        x(Varaible): Input of HardSwish operator.
```
Comment (baiyfbupt) [inline python/paddle/fluid/layers/nn.py:13109]: Input of HardSwish operator->Input of hard_swish layer

File: paddle/fluid/operators/activation_op.cc [hunk 3]
```diff
@@ -587,6 +587,32 @@ Swish Activation Operator.
   }
 };
 
+class HardSwishOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  void Make() override {
+    AddInput("X", "Input of HardSwish operator");
+    AddOutput("Out", "Output of HardSwish operator");
+    AddAttr<float>("threshold", "The threshold value of Relu6")
```
Comment (baiyfbupt) [inline paddle/fluid/operators/activation_op.cc]: remove `Relu6`?

File: python/paddle/fluid/layers/nn.py [hunk 4]
```diff
@@ -12811,3 +12812,38 @@ def shard_index(input, index_num, nshards, shard_id, ignore_value=-1):
...
+        threshold(float): The threshold value of Relu6. Default:threshold=6.0
```
Comment (baiyfbupt) [inline python/paddle/fluid/layers/nn.py]: remove `Relu6`?

File: python/paddle/fluid/layers/nn.py [hunk 5]
```diff
@@ -12811,3 +12812,38 @@ def shard_index(input, index_num, nshards, shard_id, ignore_value=-1):
...
+        scale(float): The scale parameter for the input. Default:scale=6.0
+        offset(float): The offset parameter for the input. Default:offset=3.0
+        name(str|None): A name for this layer(optional). If set None, the layer
+                        will be named automatically.
+
+    Returns:
+        Variable: Output of HardSwish operator.
```
Comment (baiyfbupt) [inline python/paddle/fluid/layers/nn.py]: HardSwish operator -> hard_swish layer

File: paddle/fluid/operators/activation_op.cc [hunk 6]
```diff
@@ -587,6 +587,32 @@ Swish Activation Operator.
...
+        .SetDefault(6.0f);
+    AddAttr<float>("scale", "The scale parameter for the input")
```
Comment (SunGaofeng) [inline paddle/fluid/operators/activation_op.cc]: scale parameter is the attribute of op, not the attribute of input. So it's not precise to say "parameter for the input", maybe "The scale parameter of HardSwish operator" is better

File: python/paddle/fluid/layers/nn.py (hunk 5 above)
Comment (SunGaofeng) [reply python/paddle/fluid/layers/nn.py]: please add "same shape as x" to show that output shape is the same as x

File: paddle/fluid/operators/activation_op.cc (hunk 3 above)
Comment (huangjun12) [reply paddle/fluid/operators/activation_op.cc]: thank you for your suggestions!

File: paddle/fluid/operators/activation_op.cc (hunk 6 above)
Comment (huangjun12) [reply paddle/fluid/operators/activation_op.cc]: thank you for your suggestions!

**Comment by huangjun12** [comment]
image

**Review by heavengate — COMMENTED** [review]
Please add API test in test_layers.py, which is required by converage CI

File: python/paddle/fluid/layers/nn.py (hunk 2 above)
Comment (huangjun12) [reply python/paddle/fluid/layers/nn.py:13109]: This refers to the style of hard_sigmoid op, relu6 op, etc. Thanks~

**Review by heavengate — APPROVED** [review]
LG API

**Review by wanghaoshuang — APPROVED** [review]
LGTM

**Review by shippingwang — APPROVED** [review]

**Pull request merged**